tests/core/test_task.py

```diff
@@ -364,13 +364,6 @@ def test_create_task_with_task_run_name(self):
 def test_task_has_logger():
     t = Task()
     assert isinstance(t.logger, logging.Logger)
-    assert t.logger.name == "prefect.Task"
```

Comment:
What is the new logger name if you're outside a flow?

Reply:
Just `prefect`. It seems `prefect.context.logger` isn't modified at all when outside a flow.
We can modify it to use `prefect.Task` if the current name is less informative than `prefect.Task`, like plain `prefect`. And use the context supplied logger otherwise.

Reply:
I'm a little concerned about breaking compatibility for people who are relying on it being `prefect.Task`, let me check with the rest of the team.

Reply:
Sure, I don't know what I was thinking deleting test to be honest :sweat_smile:. Anyways, I'm thinking something like this would be compatible with the tests I removed?
```python
    @instance_property
    def logger(self) -> logging.logging.Logger:
        # If metadata about a task exists in context, then the logger must be modified
        if prefect.context.get("task_id"):
            return prefect.context.get("logger")

        return logging.get_logger(self.name)
```

Reply:
That seems pretty reasonable. I'll check back again once I see what other people think about it.